packages/nodes-base/nodes/Hubspot/Hubspot.node.ts

```diff
@@ -935,8 +1026,206 @@ export class Hubspot implements INodeType {
+								properties: getProperties(index + batchStart),
+							}),
+							)
+								.filter(element => element.id === undefined)
+								.map(element => ({ properties: element.properties })),
+						};
+
+						const createEndpoint = `/crm/v3/objects/${customObjectType}/batch/create`;
+						const updateEndpoint = `/crm/v3/objects/${customObjectType}/batch/update`;
+
+						const createResponse = createBody.inputs.length > 0
+							? await hubspotApiRequest.call(this, 'POST', createEndpoint, createBody)
+							: undefined;
+
+						const updateResponse = updateBody.inputs.length > 0
+							? await hubspotApiRequest.call(this, 'POST', updateEndpoint, updateBody)
+							: undefined;
+
+						errors.push(...(createResponse ? getErrorsFromBatchResponse(createResponse) : []));
+						errors.push(...(updateResponse ? getErrorsFromBatchResponse(updateResponse) : []));
+
+						return [...(createResponse?.results || []), ...(updateResponse?.results || []), ...errors];
+					}
+					return [];
+
+				} catch (error) {
+					if (this.continueOnFail()) {
+						if (resource === 'customObject') {
+							const errorDetails: Record<string, unknown> = {
+								httpCode: (error as JsonObject).httpCode,
+								timestamp: (error as JsonObject).timestamp,
+								message: (error as JsonObject).message,
+							};
+
+							try {
+								const cause = (error as NodeApiError).cause as Error;
+								errorDetails.response = cause?.message;
+								const causeJsonString = cause.message?.split(' - ')[1];
+								errorDetails.response = causeJsonString;
+								const responseJson = JSON.parse(causeJsonString);
+								errorDetails.response = responseJson;
+
+								// This does not always work, because sometimes the response contains a javascript object as a string and not JSON
+								const parseMessage = (errorMessage: string) => {
+									try {
+										const responseMessageText = errorMessage?.slice(0, errorMessage?.indexOf(': '));
+										const responseMessageDataString = errorMessage?.slice(errorMessage?.indexOf(': ') + 2);
+										const responseMessageData = JSON.parse(responseMessageDataString);
+										const responseMessageObject = {
+											text: responseMessageText,
+											data: responseMessageData,
+										};
+
+										responseMessageObject.data.map((data: { message?: string } & unknown) => typeof data.message === 'string' ? { ...data, message: parseMessage(data.message) } : data);
+										return responseMessageObject;
+									} catch (e) {
+										return errorMessage;
+									}
+								};
+
+								const errorMessage = (errorDetails.response as { message: string })?.message;
+								(errorDetails.response as { message: unknown }).message = parseMessage(errorMessage);
+							} catch (e) { }
+
+							return [{ error: (error as JsonObject).message, errorDetails }];
+						}
+						return [{ error: (error as JsonObject).message }];
+					}
+					throw error;
```

Comment:
I want to return some additional details such as the id of the failed object, because just the error message returned by HubSpot is often generic. When `continueOnFail` is disabled, this is fine, because the whole response is displayed inside the spoiler. If `continueOnFail` is enabled only the message is shown and there is no way to retrieve further details about the error for logging.

For example any error during an update returns the message `Bad request - please check your parameters`. The response from Hubspot contains a lot of additional information, to get and display that in a sane way it needs to be parsed.

The parsing can probably simplified a bit, but it should work for now.